### PR TITLE
testing NSTemplateTier update progress

### PR DESCRIPTION
**Pull request by xcoulon**
E2E Tests for https://github.com/codeready-toolchain/host-operator/pull/240

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
for now I'm getting the following kind of errors in the host-operator logs:
```
{"level":"error","ts":1594216926.8537183,
"logger":"controller_nstemplatetier",
"msg":"unable to ensure TemplateRequestUpdate resource after NSTemplateTier changed.",
"Reconciler":"secondary",
"Request.Namespace":"xcoulon-host-08155246",
"Request.Name":"cheesecake",
"error":"unable to get active TemplateUpdateRequests: unable to increment the 'success' status counter after 'cheesecakelover10' completed: Operation cannot be fulfilled on nstemplatetiers.toolchain.dev.openshift.com \"cheesecake\": the object has been modified; please apply your changes to the latest version and try again",
"errorVerbose":"Operation cannot be fulfilled on nstemplatetiers.toolchain.dev.openshift.com \"cheesecake\": the object has been modified; please apply your changes to the latest version and try again\nunable to increment the 'success' status counter after 'cheesecakelover10' completed\ngithub.com/codeready-toolchain/host-operator/pkg/controller/nstemplatetier.(*baseReconciler).activeTemplateUpdateRequests\n\t/Users/xcoulon/code/go/src/github.com/codeready-toolchain/host-operator/pkg/controller/nstemplatetier/nstemplatetier_controller.go:321\ngithub.com/codeready-toolchain/host-operator/pkg/controller/nstemplatetier.(*baseReconciler).ensureTemplateUpdateRequest\n\t/Users/xcoulon/code/go/src/github.com/codeready-toolchain/host-operator/pkg/controller/nstemplatetier/nstemplatetier_controller.go:216
...
```
and has a result, the test fails when asserting the `total` and `success` counters 😬

**Comment by xcoulon** [comment]
@alexeykazakov @MatousJobanek PR is ready for review. As discussed during our meeting, I refactored the initial PR by using a single controller, and the `status.updates` do NOT include the `totalProcessed` counter because due to race conditions (I assume), the e2e tests fail to assert the exact value of this counter (I would sometimes have `12` instead of `11` for example). However, if we decide that the *exact* number of processed resources is less important than being able to track the progress, then I can easily restore this counter (or in a separate PR 😅 )

File: test/e2e/nstemplatetier_test.go [hunk 1]
```diff
@@ -100,7 +101,7 @@ func TestUpdateNSTemplateTier(t *testing.T) {
 
 	count := 2*MaxPoolSize + 1
 	ctx, awaitility := testsupport.WaitForDeployments(t, &toolchainv1alpha1.NSTemplateTier{})
-	defer ctx.Cleanup()
+	// defer ctx.Cleanup()
```
Comment (alexeykazakov) [inline test/e2e/nstemplatetier_test.go:104]: Why not performing cleanup?

Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go:104]: it's a left-over: at some point I commented out this line so I could see how the resources looked like at the end of the test. This was not intended to be part of the final PR 🤦‍♂️

Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go]: reverted in a21cbb0

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ func TestE2EFlow(t *testing.T) {
 	// given
 	// full flow from usersignup with approval down to namespaces creation
 	ctx, awaitility := testsupport.WaitForDeployments(t, &toolchainv1alpha1.UserSignupList{})
-	defer ctx.Cleanup()
+	// defer ctx.Cleanup()
```
Comment (alexeykazakov) [inline test/e2e/e2e_test.go:34]:
```suggestion
	defer ctx.Cleanup()
```

Comment (xcoulon) [reply test/e2e/e2e_test.go]: oh doh! 🤦‍♂️ 
fixed in 5530e15

File: test/e2e/nstemplatetier_test.go [hunk 3]
```diff
@@ -214,8 +245,9 @@ func verifyResourceUpdates(t *testing.T, awaitility *wait.Awaitility, syncIndexe
 
 	templateRefs := tiers.GetTemplateRefs(hostAwaitility, tier.Name)
 	require.NoError(t, err)
-	checks, err := tiers.NewChecks(aliasTierName) // here we need to use the `advanced` tier name so we can init the tier checks :|
-	require.NoError(t, err)
+	// TODO: restore
```
Comment (alexeykazakov) [inline test/e2e/nstemplatetier_test.go:248]: Did you forget to restore this one?

File: test/e2e/nstemplatetier_test.go [hunk 4]
```diff
@@ -232,9 +264,10 @@ func verifyResourceUpdates(t *testing.T, awaitility *wait.Awaitility, syncIndexe
 		)
 		require.NoError(t, err)
 		require.NotNil(t, userAccount)
-		nsTemplateSet, err := memberAwaitility.WaitForNSTmplSet(usersignup.Status.CompliantUsername)
-		require.NoError(t, err)
-		tiers.VerifyGivenNsTemplateSet(t, memberAwaitility, nsTemplateSet, checks, templateRefs)
+		// TODO: restore these assertions (commented out for now because they take a bit of time to complete)
```
Comment (alexeykazakov) [inline test/e2e/nstemplatetier_test.go:267]: And this too?

File: test/e2e/nstemplatetier_test.go [hunk 5]
```diff
@@ -214,8 +245,9 @@ func verifyResourceUpdates(t *testing.T, awaitility *wait.Awaitility, syncIndexe
 
 	templateRefs := tiers.GetTemplateRefs(hostAwaitility, tier.Name)
 	require.NoError(t, err)
-	checks, err := tiers.NewChecks(aliasTierName) // here we need to use the `advanced` tier name so we can init the tier checks :|
-	require.NoError(t, err)
+	// TODO: restore
+	// checks, err := tiers.NewChecks(aliasTierName) // here we need to use the `advanced` tier name so we can init the tier checks :|
+	// require.NoError(t, err)
```
Comment (alexeykazakov) [inline test/e2e/nstemplatetier_test.go:250]:
```suggestion
	checks, err := tiers.NewChecks(aliasTierName) // here we need to use the `advanced` tier name so we can init the tier checks :|
	require.NoError(t, err)
```

File: test/e2e/nstemplatetier_test.go [hunk 6]
```diff
@@ -232,9 +264,10 @@ func verifyResourceUpdates(t *testing.T, awaitility *wait.Awaitility, syncIndexe
 		)
 		require.NoError(t, err)
 		require.NotNil(t, userAccount)
-		nsTemplateSet, err := memberAwaitility.WaitForNSTmplSet(usersignup.Status.CompliantUsername)
-		require.NoError(t, err)
-		tiers.VerifyGivenNsTemplateSet(t, memberAwaitility, nsTemplateSet, checks, templateRefs)
+		// TODO: restore these assertions (commented out for now because they take a bit of time to complete)
+		// nsTemplateSet, err := memberAwaitility.WaitForNSTmplSet(usersignup.Status.CompliantUsername)
+		// require.NoError(t, err)
+		// tiers.VerifyGivenNsTemplateSet(t, memberAwaitility, nsTemplateSet, checks, templateRefs)
```
Comment (alexeykazakov) [inline test/e2e/nstemplatetier_test.go:270]:
```suggestion
		nsTemplateSet, err := memberAwaitility.WaitForNSTmplSet(usersignup.Status.CompliantUsername)
		require.NoError(t, err)
		tiers.VerifyGivenNsTemplateSet(t, memberAwaitility, nsTemplateSet, checks, templateRefs)
```

File: test/e2e/nstemplatetier_test.go (hunk 3 above)
Comment (alexeykazakov) [reply test/e2e/nstemplatetier_test.go]: Restored.

File: test/e2e/nstemplatetier_test.go (hunk 4 above)
Comment (alexeykazakov) [reply test/e2e/nstemplatetier_test.go]: Restored.

**Pull request merged**